programs/stake/src/stake_state.rs

```diff
@@ -755,12 +755,15 @@ impl Stake {
         clock: &Clock,
         stake_history: &StakeHistory,
         config: &Config,
+        can_reverse_deactivation: bool,
     ) -> Result<(), StakeError> {
         // If stake is currently active:
         if self.stake(clock.epoch, Some(stake_history), true) != 0 {
             // If pubkey of new voter is the same as current,
-            // and we have not deactivated any stake yet, we clear deactivation
-            if self.delegation.voter_pubkey == *voter_pubkey
+            // and we are scheduled to start deactivating this epoch,
+            // we rescind deactivation
+            if can_reverse_deactivation
```

Comment:
NIT: If this condition is last, the post-activation removal patch will be cleaner :wink:

Reply:
Ah - how long after is the post-removal patch?

I was thinking about how ugly the code would be if you kept patching this way using features. 

Is the idea that one would have a new stake program Ver B, and a "global feature" that toggles between the Ver A various patches and a new one?

Reply:
A bit confused how much the team would like to respect the tx replayability using the newest version of the client, and how much they'd prefer to consider that legacy tx history that can't be replayed.

Reply:
> A bit confused how much the team would like to respect the tx replayability using the newest version of the client, and how much they'd prefer to consider that legacy tx history that can't be replayed.

After a feature has been active for a major release or two we'll remove the feature from the master branch when it starts to get in the way, or if somebody is bored.

Reply:
Hmm, I see, so you'd have to replay past txs with an older validator release then. Not that anyone actually does that.